modules/Twitter/src/main/java/totetmatt/gephi/twitter/networklogic/FullSmartNetwork.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package totetmatt.gephi.twitter.networklogic;
+
+import java.awt.Color;
+import java.util.Arrays;
+import org.gephi.graph.api.Edge;
+import org.gephi.graph.api.Node;
+import org.openide.util.lookup.ServiceProvider;
+import twitter4j.HashtagEntity;
+import twitter4j.MediaEntity;
+import twitter4j.Status;
+import twitter4j.SymbolEntity;
+import twitter4j.URLEntity;
+import twitter4j.UserMentionEntity;
+
+/**
+ *
+ * @author totetmatt
+ */
+@ServiceProvider(service = Networklogic.class)
+public class FullSmartNetwork extends Networklogic {
+
+    public FullSmartNetwork() {
+
+    }
+
+    @Override
+    public void onStatus(Status status) {
+        processStatus(status, null);
+    }
+
+    public void processStatus(Status status, Node retweetUser) {
+        Node tweet = graphController.getGraphModel().getGraph().getNode("" + status.getId());
```

Comment:
I don't know how much of a problem this is, but it is possible that the graph could be modified in between some of these statements. For example the node representing the tweet could exist, but then be deleted before the createLink(user,tweet) call. To avoid this, you can add
``graphController.getGraphModel().getGraph().writeLock()``
at the beginning of the function and
```graphController.getGraphModel().getGraph().writeUnlock()``
at the end. This will prevent changes to the graph while you are updating it.